### PR TITLE
Update Psych load  RBI for version 4

### PR DESCRIPTION
**Pull request by miry**
There are new changes in `Psych.load`. It is pointed to `safe_load`.

### Motivation
Currently `YAML.load("---", permitted_classes: [Symbol, Date, Time])` is not valid and produce the error: `Unrecognized keyword argument permitted_classes passed for method Psych.load`

### Test plan

See included automated tests.

### CODE REVIEW

File: rbi/stdlib/psych.rbi [hunk 1]
```diff
@@ -271,128 +271,99 @@ module Psych
...
-  #
-  # NOTE: This method \*should not\* be used to parse untrusted documents, such
-  # as [`YAML`](https://docs.ruby-lang.org/en/2.7.0/YAML.html) documents that
-  # are supplied via user input. Instead, please use the
-  # [`safe_load`](https://docs.ruby-lang.org/en/2.7.0/Psych.html#method-c-safe_load)
-  # method.
   sig do
     params(
       yaml: T.any(String, StringIO, IO),
-      legacy_filename: Object,
```
Comment (paracycle) [inline rbi/stdlib/psych.rbi:321]: For backwards compatibility, I think this should stay in place and not be removed.

File: rbi/stdlib/psych.rbi [hunk 2]
```diff
@@ -271,128 +271,99 @@ module Psych
...
-  # Psych.safe_load("---\n foo: bar")                         # => {"foo"=>"bar"}
-  # Psych.safe_load("---\n foo: bar", symbolize_names: true)  # => {:foo=>"bar"}
-  # ```
   sig do
     params(
       yaml: T.any(String, StringIO, IO),
-      legacy_permitted_classes: Object,
-      legacy_permitted_symbols: Object,
-      legacy_aliases: Object,
-      legacy_filename: Object,
```
Comment (paracycle) [inline rbi/stdlib/psych.rbi:395]: Same for these. These should not be removed.

**Review by paracycle — CHANGES_REQUESTED** [review]

**Review by Morriar — COMMENTED** [review]
I'm not sure the documentation changes are in the format we expect.

We generally run them through this script: https://github.com/sorbet/sorbet/blob/master/rbi/tools/sync-rdoc.rb:

```
$ ./rbi/tools/sync-rdoc.rb rbi/stdlib/psych.rbi
```

**Comment by elliottt** [comment]
I'm happy to run this PR on Stripe's CI once the comments from @paracycle and @Morriar have been addressed 👍

**Comment by miry** [comment]
Thanks for the `$ ./rbi/tools/sync-rdoc.rb rbi/stdlib/psych.rbi`. using it I found it looks like RDoc for Psych is not correct :)
I am going to tune it a bit myself.

File: rbi/stdlib/psych.rbi (hunk 1 above)
Comment (miry) [reply rbi/stdlib/psych.rbi:321]: fixed

**Comment by miry** [comment]
@Morriar @paracycle Thank you for the feedback. Cleaned up, and only add missing attributes.

File: rbi/stdlib/psych.rbi (hunk 2 above)
Comment (miry) [reply rbi/stdlib/psych.rbi]: reverted

**Review by elliottt — CHANGES_REQUESTED** [review]
Happy to test on Stripe's codebase once the test failures are addressed 👍

**Comment by miry** [comment]
@elliottt Do you know what would be the best way to run test on local environment.

I tried `bazel test`

```shell
$ bazel test                  
INFO: Invocation ID: f29f24e2-b86d-4b3d-a4e5-055879303dd7
Loading: 
Loading: 0 packages loaded
INFO: Build options --copt, --define, and --linkopt have changed, discarding analysis cache.
Analyzing: 0 targets (0 packages loaded, 0 targets configured)
INFO: Analyzed 0 targets (0 packages loaded, 0 targets configured).
INFO: Found 0 test targets...
[0 / 1] [Prepa] BazelWorkspaceStatusAction stable-status.txt
INFO: Elapsed time: 0.127s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
ERROR: No test targets were found, yet testing was requested
INFO: Build completed successfully, 1 total action
```

Also tried to run explicit command for buildkite that failing - without success to finish build part.

**Comment by miry** [comment]
@elliottt The tests are passing. It is ready for next stage.

**Comment by elliottt** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_L1ZOjD3miIQa8m
→ https://go/builds/bui_L1ZOte0cKJxArV

**Review by elliottt — APPROVED** [review]
Thank you!

**Pull request merged**